### DISCUSSION TITLE
`break-words` class is great but not enough

### DISCUSSION THREAD

**Discussion by lucascaton** [tailwindlabs/tailwindcss#2213] [Ideas]
Hi there.

It'd be nice to have a single Tailwind class to apply both `overflow-wrap: break-word` and `word-break: break-word` declarations.
The [break-words](https://tailwindcss.com/docs/word-break#break-words) class is great but doesn't cover all scenarios (sorry if I'm missing something). Here's an example:

* Using the `break-words` class:

  ![1](https://user-images.githubusercontent.com/86755/90846194-028e1480-e3ab-11ea-9ff9-9f1acbec4a4b.png)

* Using `overflow-wrap: break-word` and `word-break: break-word` declarations:

  ![2](https://user-images.githubusercontent.com/86755/90846198-0457d800-e3ab-11ea-950d-a88973b0cda6.png)

---

Example:

```html
<div class="bg-gray-200 h-screen">
  <main class="container mx-auto bg-white h-screen">
    <table class="table-auto w-full mx-auto border border-gray-200 rounded">
      <tbody class="bg-gray-400">
        <tr>
          <td class="p-6">Column</td>
          <td class="p-6">Column</td>
          <td class="p-6">Column</td>
          <td class="p-6">Column</td>
          <td class="p-6">Column</td>
          <td class="p-6">Column</td>

          <td class="p-6 break-words foobar">
            https://looooooooooooooooooooooooooooooooooooooooong-url.com -
            Preserve whole words.
            Preserve whole words.
            Preserve whole words.
            Preserve whole words.
            Preserve whole words.
            Preserve whole words.
          </td>
        </td>
      </tbody>
    </table>
  </main>
</div>
```

```css
/* ↓ Uncomment the following lines to solve the issue ↓ */

/* td.foobar {
  overflow-wrap: break-word;
  word-break: break-word;
} */
```

([Codepen with the same example](https://codepen.io/lucascaton/pen/zYqoogX))

Please let me know your thoughts 🙂

**Comment by adamwathan** [comment]
Shitty one, what do you think the best solution is? `word-break: break-word` is deprecated and recommended against by MDN, but `overflow-wrap: anywhere` isn't supported in Safari.

You don't need `overflow-wrap: break-word` if using `word-break: break-word` I don't _think_ anyways, at least in my testing. These are some of the most confusing CSS properties in existence :/

**Reply by rightaway** [reply]
@adamwathan Is there any downside to adding `overflow-wrap: anywhere` and `word-break: break-word` to `break-words` utility? The first would be for all browsers except Safari and the second deprecated one would be for Safari. Because otherwise you have to add these styles manually to the markup wherever you use `break-words`.

**Comment by lucascaton** [comment]
> `word-break: break-word` is deprecated

Hmmm, you're right 😞 ([MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/word-break))

> what do you think the best solution is?

It looks like `overflow-wrap: anywhere` that you suggested solves the issue but as you said, [Safari doesn't support it](https://caniuse.com/mdn-css_properties_overflow-wrap_anywhere).

Are you planning to create a Tailwind class for `overflow-wrap: anywhere`?

Thanks, Adam!

**Reply by KBM90** [reply]
Right , I find **overflow-wrap:anywhere**   most efficient , for my case I used  with the table data after a bunch of tries with normal and ellipsis tailwind's properties

**Reply by lucascaton** [reply]
Update: [Safari supports it now](https://caniuse.com/mdn-css_properties_overflow-wrap_anywhere) 🎉

**Reply by borisghidaglia** [reply]
Hey 👋
> Hmmm, you're right 😞 ([MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/word-break))

Sorry if I'm missing something, but is `word-break: break-word` still deprecated? 
The link you shared here doesn't show any deprecation anymore, right?

**Comment by AdamZaczek** [comment]
Bump - overflow-wrap:anywhere is needed!

**Comment by FiveOFive** [comment]
According to https://github.com/mdn/browser-compat-data/issues/17012 overflow-wrap: anywhere is now supported in Safari since 15.4.

**Comment by Jakeduncan00** [comment]
.one {
  word-break: break-all
}
.two {
  word-wrap: break-word;
}
<div style="width:56mm; font-family:Arial; border: 2px solid tomato; background: gold;">
  <b>word-break: break-all:</b>
  <p>Your selected shipping was <span class=one>UK_RoyalMailSecondClassStandard</span>.</p>
  <p>Your selected shipping was <span class=one>Signed For 1st Class</span>.</p>
  <b>word-wrap: break-word:</b>
  <p>Your selected shipping was <span class=two> UK_RoyalMailSecondClassStandard</span>.</p>
  <p>Your selected shipping was <span class=two>Signed For 1st Class</span>.</p>
</div

[Krowd Darden
](https://www.krowddarden.net/)

**Comment by rap2hpoutre** [comment]
> According to https://github.com/mdn/browser-compat-data/issues/17012 overflow-wrap: anywhere is now supported in Safari since 15.4.

@adamwathan Let's go then? 

Also: https://caniuse.com/mdn-css_properties_overflow-wrap_anywhere

**Comment by mortennajbjergjp** [comment]
The example from @lucascaton shows a table. I am experiencing a similar issue using grid.
This is the solution I use for now:

```
@tailwind base;
@tailwind components;
@tailwind utilities;

@layer utilities {
  .overflow-anywhere {
    overflow-wrap: anywhere;
  }
}
```

**Reply by shrekuu** [reply]
I guess @adamwathan would choose `break-anywhere` as the class name. 😆

**Comment by Blackklegend** [comment]
Bump - overflow-wrap:anywhere pls

**Comment by adamwathan** [comment]
Added support for `overflow-wrap: anywhere` to my list of stuff to look into!

In the mean time don't forget you can always use arbitrary properties:

```html
<div class="[overflow-wrap:anywhere]">
```

**Comment by evybauer** [comment]
`break-all ` should solve the issue
https://tailwindcss.com/docs/word-break#break-all

**Reply by shrekuu** [reply]
Well, it doesn't. 

We just want to break words that are too long(more than one line), and keep these short ones(less than one line).

I created a demo here. https://play.tailwindcss.com/fhv0vZlsea

We want the **good old `word-break: break-word`** one. I never know why this is deprecated without a ready alternative.

Also, check out this answer. https://stackoverflow.com/a/75221783

**Comment by stromseng** [comment]
Hey, i saw this was being looked into, but haven't seen any updates on it. Just stumbled upon this myself, and would greatly appreciate support for `overflow-wrap: anywhere`

**Reply by DePasqualeOrg** [reply]
Same here. I found this by searching for "tailwind break-words not working" and am using `[overflow-wrap:anywhere]` for now.

**Reply by kvtemadden** [reply]
Snap, prefer to avoid arbitrary values as much as possible. Hoping to see it at some point soon.

**Reply by johanbook** [reply]
Ran into this issue as well and would love seeing it as part of Tailwind

**Comment by agjs** [comment]
@adamwathan Any news on this? So many people are asking for it...

**Comment by OmkarJ13** [comment]
Bump. This class is much needed

**Comment by danielraab** [comment]
Same discussion here: https://github.com/tailwindlabs/tailwindcss/discussions/12127
with a pull request: https://github.com/tailwindlabs/tailwindcss/pull/12128

**Comment by lucascaton** [comment]
This should be fixed in Tailwind CSS 4.1 🎉

https://tailwindcss.com/blog/tailwindcss-v4-1#fine-grained-text-wrapping-with-overflow-wrap
